src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1548,7 +1559,12 @@ private static string ReadError(StreamReader reader)
             // Blocking read from StdError stream
             string error = reader.ReadLine();
 
-            if (string.IsNullOrEmpty(error) ||
+            if (error == null)
+            {
+                return error;
```

Comment:
Why did you decide to return null here and throw in the calling function? Why not just throw here and make the ReadError() function always return non-null strings?